ompi/mpi/c/info_f2c.c

```diff
@@ -65,10 +65,6 @@ MPI_Info MPI_Info_f2c(MPI_Fint info)
         return MPI_INFO_ENV;
     }
 
-    if (MPI_PARAM_CHECK) {
```

Comment:
I checked to make sure that we don't have to call `ompi_mpi_instance_retain` and found that it's called somewhere from within `MPI_INFO_CREATE_ENV` (in `ompi_info_allocate`). So from there on all info functions are safe. Maybe we should document that for the next poor guy trying to understand why we don't explicitly retain the environment here?

Reply:
added some verbiage about this.  Actually it's an application error if it tries to get the fortran handle for MPI_INFO_ENV prior to calling MPI_init.